exporter/exporterhelper/common.go

```diff
@@ -111,7 +113,47 @@ func WithQueue(config QueueSettings) Option {
 			}
 			return
 		}
-		o.queueSender = newQueueSender(config, o.set, o.signal, o.marshaler, o.unmarshaler)
+		var queue internal.Queue[Request]
+		set := internal.QueueCreateSettings[Request]{
+			CreateSettings: o.set,
+			Sizer:          &internal.RequestSizer[Request]{},
+			Capacity:       config.QueueSize,
+			DataType:       o.signal,
+		}
+		if config.StorageID != nil {
+			queue = internal.NewPersistentQueue[Request](set, *config.StorageID, o.marshaler, o.unmarshaler)
+		} else {
+			queue = internal.NewBoundedMemoryQueue[Request](set)
+		}
+		o.queueSender = newQueueSender(queue, o.set, config.NumConsumers)
+	}
+}
+
+// WithRequestQueue enables queueing for an exporter.
+// This option should be used with the new exporter helpers New[Traces|Metrics|Logs]RequestExporter.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+func WithRequestQueue(cfg configqueue.QueueConfig, queueFactory QueueFactory) Option {
```

Comment:
Instead of having a new public `QueueFactory` what about we change this to `func WithRequestQueue(queue Queue, numConsumers int)`

Reply:
Not sure how to pass `QueueCreateSettings` then. Should we move it to the Start method?

Reply:
`QueueCreateSettings` - What is in there that user doesn't have access to or you think is "duplicate"?

Reply:
Yes, we would need to ask the user to pass at least `component.DataType` and `exporter.CreateSettings` to the persistent queue constructor in addition to the exporter helper itself